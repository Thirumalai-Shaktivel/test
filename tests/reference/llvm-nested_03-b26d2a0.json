{
    "basename": "llvm-nested_03-b26d2a0",
    "cmd": "lfortran --show-llvm {infile} -o {outfile}",
    "infile": "tests/../integration_tests/nested_03.f90",
    "infile_hash": "5ce081ba0592ea127cc510ce95402df1b8d0d9fd87200417737ad294",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "llvm-nested_03-b26d2a0.stdout",
    "stdout_hash": "154dcc154b8964111d0b467ef58f51bc0554d276c30bebbed881e4fb",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}