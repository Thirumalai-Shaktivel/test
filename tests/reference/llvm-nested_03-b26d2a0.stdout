; ModuleID = 'LFortran'
source_filename = "LFortran"

%b = type { i32 }

@b_rtd = global %b zeroinitializer
@0 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@1 = private unnamed_addr constant [4 x i8] c"%d\0A\00", align 1
@2 = private unnamed_addr constant [4 x i8] c"b()\00", align 1
@3 = private unnamed_addr constant [4 x i8] c"%s\0A\00", align 1

define void @__module_nested_03_a_c() {
.entry:
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @0, i32 0, i32 0), i32 5)
  %0 = load i32, i32* getelementptr inbounds (%b, %b* @b_rtd, i32 0, i32 0), align 4
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @1, i32 0, i32 0), i32 %0)
  ret void
}

declare void @_lfortran_printf(i8*, ...)

define void @__module_nested_03_a_b() {
.entry:
  %x = alloca i32, align 4
  store i32 6, i32* %x, align 4
  %0 = load i32, i32* %x, align 4
  store i32 %0, i32* getelementptr inbounds (%b, %b* @b_rtd, i32 0, i32 0), align 4
  call void (i8*, ...) @_lfortran_printf(i8* getelementptr inbounds ([4 x i8], [4 x i8]* @3, i32 0, i32 0), i8* getelementptr inbounds ([4 x i8], [4 x i8]* @2, i32 0, i32 0))
  call void @__module_nested_03_a_c()
  ret void
}

define i32 @main() {
.entry:
  call void @__module_nested_03_a_b()
  ret i32 0
}

